src/ar_storage.erl

```diff
@@ -29,16 +29,17 @@ start() ->
 
 %% @doc Ensure that all of the relevant storage directories exist.
 ensure_directories() ->
-	filelib:ensure_dir(?TX_DIR),
-	filelib:ensure_dir(?BLOCK_DIR),
-	filelib:ensure_dir(?BLOCK_ENC_DIR),
-	filelib:ensure_dir(?WALLET_LIST_DIR),
-	filelib:ensure_dir(?HASH_LIST_DIR),
+	DataDir = ar_meta_db:get(data_dir) ++ "/",
+	filelib:ensure_dir(DataDir ++ ?TX_DIR ++ "/"),
+	filelib:ensure_dir(DataDir ++ ?BLOCK_DIR ++ "/"),
+	filelib:ensure_dir(DataDir ++ ?BLOCK_ENC_DIR ++ "/"),
+	filelib:ensure_dir(DataDir ++ ?WALLET_LIST_DIR ++ "/"),
+	filelib:ensure_dir(DataDir ++ ?HASH_LIST_DIR ++ "/"),
```

Comment:
Shouldn't also the WALLET_DIR be created here?

Reply:
`ar_wallet:new_keyfile` takes care of it.

Reply:
✅